App.js

```diff
@@ -1,17 +1,51 @@
-import React, { useState } from "react";
+import React, { useState, useEffect, useCallback } from "react";
 import { NavigationContainer } from "@react-navigation/native";
+import * as SplashScreen from "expo-splash-screen";
 
 import AuthContext from "./app/auth/context";
 import navigationTheme from "./app/navigation/navigationTheme";
 import AuthNavigator from "./app/navigation/AuthNavigator";
 import AppNavigator from "./app/navigation/AppNavigator";
+import asyncStorage from "./app/store/asyncStorage";
+import { USER_OBJECT_KEY } from "./app/store/constants";
 
 export default function App() {
   const [user, setUser] = useState();
+  const [isReady, setIsReady] = useState(false);
+
+  const restoreUserSession = async () => {
+    const userObj = await asyncStorage.getDataObject(USER_OBJECT_KEY);
+    if (!userObj) return;
+    setUser(userObj);
+  };
+
+  useEffect(() => {
+    async function prepare() {
+      try {
+        await SplashScreen.preventAutoHideAsync();
+        await restoreUserSession();
+      } catch (error) {
+        console.log("Error loading app", error);
+      } finally {
+        setIsReady(true);
+      }
```

Comment:
Do we need Finally? Because incase of error we should display error screen and having boolean for ready will not distinguish whether it is succeeded or failed?

How about we initialize loadingState which can have values like [Succeeded, Failed, Loading, Idle]
So basically 
```
// Some other file 
export enum AsyncStatus {
  Idle = 'Idle',
  Loading = 'Loading',
  Succeeded = 'Succeeded',
  Failed = 'Failed',
}


// This file
const [appStatus, setAppStatus] = useState(Async.Idle);
....
try {
  ...
 setAppStatus(Async.Succeeded)
} catch(error) {
 setAppStatus(Async.Failed)
}
...
// Just demonstrating what we should render
switch(appStatus) {
 case Async.Idle: return null;
 case Async.Failed: return <ErrorComponent />;
 case Async.Succeeded: return <ActualComponent />;
 case Async.Loading: return <SplashScreen />;
}
```

Reply:
Loved this idea.. Updated